Automattic/jetpack
# My Jetpack: Introduce main section components

This PR proposes two new fundamental layout components to our Design system and applies them to the My Jetpack Screen.

## Jetpack Admin Page

This component is basically the main container for any admin page of any plugin. It adds the Header and the Footer.

## Jetpack Admin Section

This component is the wrapper for each section (or each row) of the admin screen. The propose is that anything in the admin page should be added in one of these and this will ensure all sections have the same spacing and behave the same way in every break point.

### Options

**bgColor:** There are 2 possible background colors: white and grey

Here is how they look and behave:

![Captura de Tela 2021-11-26 às 15 22 48](https://user-images.githubusercontent.com/971483/143620967-24ca0fdd-37e7-4f6b-922c-df3656cef861.png)

![Captura de Tela 2021-11-26 às 15 33 20](https://user-images.githubusercontent.com/971483/143620980-873a966e-8eb3-46dd-b9a6-c257ef3cad08.png)


#### Changes proposed in this Pull Request:

* Adds JetpackAdminPage and JetpackAdminSection components.

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






On Storybook:

* Checkout this branch
* `jetpack build js-packages/components`
* `cd projects/js-packages/storybook`
* `pnpm run storybook:dev`
* It will open your browser
* Go to PlayGround > Admin Page and Playground > Admin Section

On a live site:

* Check out this branch or launch a JN site in this branch
* Go to Settings > Jetpack Constants and enable `JETPACK_ENABLE_MY_JETPACK`
* Go to Jetpack > My Jetpack

Question to design:

Are the options offered by these wrapper elements enough to cover all our scenarios so far? Are the spacings right? Do we need any additional option?

Looking at some of the designs, it seems that the top margin of the row with a grey background is slightly smaller than the other section. Any suggestions on how to treat this? Instead of a `bgColor` option should we have something like `Hero` or `highlight` option that would add the grey background and decrease the top margin?
